packages/sync/src/utils/buildSyncLookupSelect.ts

```diff
@@ -0,0 +1,28 @@
+import { DatabaseModel } from '@tupaia/database';
+import { COLUMNS_EXCLUDED_FROM_SYNC } from '../constants';
+
+interface Columns {
+  projectIds?: string;
+}
+
+export async function buildSyncLookupSelect(model: DatabaseModel, columns: Columns = {}) {
+  const attributes = await model.fetchFieldNames();
+  const { projectIds } = columns;
+  const table = model.databaseRecord;
+
+  return `
+    SELECT DISTINCT ON (
```

Comment:
Again this is super coupled to the fact that the query it goes with is ordered etc.

I wonder if there's some way to bring these query segments closer together?

Reply:
Actually this is totally wrong. I was using it to remove duplicates for `questions` table which skips some of the project_ids if there a question is used in multiple surveys coming from multiple projects - my apology for not catching this. I've removed the `DISTINCT ON` and aggregate the project_ids array properly for `questions` table in the subsequent PR